core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -152,19 +152,18 @@ private void needAuthentication(ContainerRequestContext request, Optional<TokenP
             try {
                 redirectForNewToken(request, refreshToken.get());
             }
-            catch (ChallengeFailedException e) {
+            catch (Exception e) {
                 LOG.debug(e, "Tokens refresh challenge has failed");
```

Comment:
Not really related but since you're already making changes in this area we could consider adding additional tokens logging, probably behind flag, to make debugging easier. Similar to Hydra's `LOG_LEAK_SENSITIVE_VALUES` https://www.ory.sh/docs/ecosystem/logging

Reply:
I agree with @s2lomon we shouldn't be logging any security tokens, passwords, or other sensitive information. Image that the logs can always be read by a malicious party and act accordingly.